src/js/wrap.cpp

```diff
@@ -678,6 +678,42 @@ namespace ccf::js
     return JS_UNDEFINED;
   }
 
+  JSValue js_rpc_set_claims_digest(
+    JSContext* ctx, JSValueConst this_val, int argc, JSValueConst* argv)
+  {
+    if (argc != 1)
+    {
+      return JS_ThrowTypeError(ctx, "Passed %d arguments but expected 1", argc);
+    }
+
+    auto rpc_ctx =
+      static_cast<enclave::RpcContext*>(JS_GetOpaque(this_val, rpc_class_id));
+
+    if (rpc_ctx == nullptr)
+    {
+      return JS_ThrowInternalError(ctx, "RPC context is not set");
+    }
+
+    size_t digest_size;
+    uint8_t* digest = JS_GetArrayBuffer(ctx, &digest_size, argv[0]);
+
+    if (!digest)
+      return JS_ThrowTypeError(ctx, "Argument must be an ArrayBuffer");
+
+    if (digest_size != ccf::ClaimsDigest::Digest::SIZE)
+      return JS_ThrowTypeError(
+        ctx, "Argument must be an ArrayBuffer of the right size");
+
+    ccf::ClaimsDigest::Digest::Representation r;
+    std::copy(digest, digest + digest_size, r.begin());
+    ccf::ClaimsDigest::Digest d;
+    d.set(std::move(r));
```

Comment:
Couldn't we use the `Sha256Hash(const CBuffer& data)` constructor here instead?

Reply:
I'd like to kill off CBuffer, so maybe a from_span() or something like that.